docs/source/tutorials/maintainers_guide.rst

```diff
@@ -0,0 +1,35 @@
+#################
+Maintainers Guide
+#################
+
+Here are some notes on how to maintain this package mostly focusing on the CI/CD workflow build on top of GitHub actions.
+
++++++++++++++++++++++++
+The pull request checks
++++++++++++++++++++++++
+
+
+The majority of the pull request checks are specified in the CI. Specifically, the `pre-merge.yaml <https://github.com/AlignmentResearch/tuned-lens/blob/improved-docs-85/.github/workflows/pre-merge.yml>`_ workflow. There are 4 major components to this workflow:
+
+1. Ensuring that the pre-commit checks configured in ``.pre-commit-config.yaml`` pass.
+2. Ensuring that the package builds correctly and the ``pytest`` tests pass on python versions 3.9 - 3.11.
+    * ``pytest`` is configured in the ``pyproject.toml``.
+3. Ensuring that this documentation builds correctly and the code within it runs including the tutorial notebooks.
+    * The documentation is built using `sphinx <https://www.sphinx-doc.org/en/master/>`_ and the tutorial notebooks are run using `nbsphinx <https://nbsphinx.readthedocs.io/en/latest>`_.
+4. Ensuring that the docker image builds correctly and uploading code coverage reports to `codecov <https://codecov.io/gh/AlignmentResearch/tuned-lens>`_.
+    * The code coverage requirements themselves are contained in ``.codecov.yml``. Importantly, the code coverage bot itself enforce these requirements, not the CI.
+
+Note that the pre-merge workflow also runs on every push to the main branch. To make sure this passes is best practice to merge main into the branch before merging your PR.
```

Comment:
It's possible to force this in the "protect your branch" settings. I think the setting called "Require branches to be up to date before merging" does this.

Reply:
Oh, thanks I didn't know about that option!